src/cocotb/clock.py

```diff
@@ -152,42 +186,84 @@ async def start(self, start_high: bool = True) -> None:
 
                 .. versionadded:: 1.3
 
+        Returns:
+            Object which can be passed to :func:`cocotb.start_soon` or ignored.
+
         .. versionchanged:: 2.0
             Removed ``cycles`` arguments for toggling for a finite amount of cyles.
-            Use ``kill()`` on the clock task instead, or implement manually.
+            Use :meth:`stop` to stop a clock from running.
```

Comment:
Maybe mention here that a stopped clock can also be restarted.

Reply:
It think this comment is more about what changed with the cycles command and that's unrelated.